framework/src/iotbus/iotbus_uart.c

```diff
@@ -176,6 +177,69 @@ int iotbus_uart_stop(iotbus_uart_context_h hnd)
 	return IOTBUS_ERROR_NONE;
 }
 
+int iotbus_uart_rxavailable(iotbus_uart_context_h hnd)
+{
+	int ret;
+	int fd;
+	struct _iotbus_uart_s *handle;
+
+	if (!hnd || !hnd->handle) {
+		return IOTBUS_ERROR_INVALID_PARAMETER;
+	}
+
+	handle = (struct _iotbus_uart_s *)hnd->handle;
+	fd = handle->fd;
+
+	ret = tccheckfifo(fd, TIOCS_AVAIL);
+	if (ret < 0) {
+		return IOTBUS_ERROR_UNKNOWN;
+	}
+
+	return IOTBUS_ERROR_NONE;
+}
+
+int iotbus_uart_txready(iotbus_uart_context_h hnd)
+{
+	int ret;
+	int fd;
+	struct _iotbus_uart_s *handle;
+
+	if (!hnd || !hnd->handle) {
+		return IOTBUS_ERROR_INVALID_PARAMETER;
+	}
+
+	handle = (struct _iotbus_uart_s *)hnd->handle;
+	fd = handle->fd;
+
+	ret = tccheckfifo(fd, TIOCS_READY);
+	if (ret < 0) {
```

Comment:
How about using specific error code like IOTBUS_ERROR_DEBICE_NOT_READY for easier debugging?

Reply:
done